### PR TITLE
Implement HasUID class to LabelledContents and Section

### PR DESCRIPTION
**Pull request by halonazhao**
In this PR:
1. LabelledContent and Section were implemented `HasUID` class. Since they both have `label` field and `label` has a UID, the HasUID directly pulls the UID from the label to attach to the Chunk.
2. UID field was added to Reference2 for future use.

Question:
Since there were some labels created for `list`, do we remove them to keep them as unlabelled contents? @JacquesCarette

### CODE REVIEW

File: code/drasil-docLang/Drasil/Sections/SpecificSystemDescription.hs [hunk 1]
```diff
@@ -172,7 +172,7 @@ dataConstraintParagraph hasUncertainty tableRef middleSent trailingSent = mkPara
 -- makes a list of references to tables takes
 -- l  list of layout objects that can be referenced
 -- outputs a sentence containing references to the layout objects 
-listofTablesToRefs :: (HasShortName l, Referable l) => [l] -> Sentence
+listofTablesToRefs :: (HasUID l, HasShortName l, Referable l) => [l] -> Sentence
```
Comment (JacquesCarette) [inline code/drasil-docLang/Drasil/Sections/SpecificSystemDescription.hs]: I think this `HasUID` constraint can be lifted now.

File: code/drasil-example/Drasil/NoPCM/GenDefs.hs [hunk 2]
```diff
@@ -45,7 +45,7 @@ roc_temp_simp_deriv_sentences = map foldlSentCol [
     QT.temp vol [makeRef2S newA3, makeRef2S newA4, makeRef2S newA5],
   genDefDesc5 density mass vol]
 
-genDefDesc1 :: (HasShortName x, Referable x) => x -> UnitalChunk -> [Sentence]
+genDefDesc1 :: (HasUID x, HasShortName x, Referable x) => x -> UnitalChunk -> [Sentence]
```
Comment (JacquesCarette) [inline code/drasil-example/Drasil/NoPCM/GenDefs.hs]: And this one too? Though we might consider making `HasUID` a pre-condition of `Referable`.

File: code/drasil-example/Drasil/SSP/Body.hs [hunk 3]
```diff
@@ -344,7 +344,7 @@ phys_sys_desc_p1 = physSystIntro slope how intrslce slice
   (S "slice base") fig_indexconv
   where how = S "as a series of" +:+ phrase slice +:+. plural element
 
-physSystIntro :: (NamedIdea a, NamedIdea b, NamedIdea c, HasShortName d, Referable d) =>
+physSystIntro :: (NamedIdea a, NamedIdea b, NamedIdea c, HasUID d, HasShortName d, Referable d) =>
```
Comment (JacquesCarette) [inline code/drasil-example/Drasil/SSP/Body.hs]: Same here.

File: code/drasil-lang/Language/Drasil/Reference.hs [hunk 4]
```diff
@@ -252,22 +252,22 @@ assumptionsFromDB am = dropNums $ sortBy (compare `on` snd) assumptions
...
-makeCite l = Reference2 Citation (refAdd l) (l ^. shortname)
+makeCite l = Reference2 (l ^. uid) Citation (refAdd l) (l ^. shortname)
 
 makeCiteS :: Citation -> Sentence
 makeCiteS = Ref2 . makeCite
 
 -- | Create a reference for a URI
 makeURI :: String -> ShortName -> Reference2
-makeURI ra sn = Reference2 URI ra sn
+makeURI ra sn = Reference2 "fixme" URI ra sn
```
Comment (JacquesCarette) [inline code/drasil-lang/Language/Drasil/Reference.hs]: I don't really want to accept a PR with a "fixme" like that in it...

File: code/drasil-example/Drasil/GamePhysics/Body.hs [hunk 5]
```diff
@@ -434,7 +437,10 @@ goal_statements_list' = map (foldlSent) [goal_statements_G_linear,
   goal_statements_G_angular, goal_statements_G_detectCollision, 
   goal_statements_G_collision]
 
-goal_statements_list = enumSimple 1 (getAcc goalStmt) goal_statements_list'
+goal_label :: Label
+goal_label = mkLabelRALst "goalGM" "goalGM"
```
Comment (JacquesCarette) [inline code/drasil-example/Drasil/GamePhysics/Body.hs]: It seems that `goal_label` (and `solution_label` and ...) are duplicated for each example. They should be pulled out into drasil-docLang.

**Comment by halonazhao** [comment]
Sorry I didn't put the priority on this because I really wanted to push on my process so that some fixes wouldn't kill my patience. Does putting labels in docLang is enough? Or even take the constructor to docLang? @JacquesCarette

**Comment by JacquesCarette** [comment]
Branch merged - should I also delete it, because the work on this (addUID) is 'done'?

**Comment by halonazhao** [comment]
Yes, please. We are done with `addUID` branch.

**Comment by JacquesCarette** [comment]
Done.

**Pull request merged**